modules/data.atmosphere/R/download.NOAA_GEFS.R

```diff
@@ -0,0 +1,260 @@
+  #each ensemble member.  Each variable gets its own matrix, which is stored in the list noaa_data.
+  for (i in 1:length(noaa_var_names)) {
+    noaa_data[[i]] = rnoaa::gefs(noaa_var_names[i], lat.in, lon.in, raw=TRUE, time_idx = 1:increments, forecast_time = forecast_hour, date=format(start_date, "%Y%m%d"))$data
+  }
+  
+  ###################################################
+  # Not all NOAA data units match the cf data standard.  In this next section, data are processed to
+  # confirm with the standard when necessary.
+  # The following is a list of variables which need to be processed:
+  # 1. NOAA's relative humidity must be converted to specific humidity
+  # 2. NOAA's measure of precipitation is the accumulation over 6 hours; cf's standard is precipitation per second
+
+  #Convert NOAA's relative humidity to specific humidity
+  humid_index = which(cf_var_names == "specific_humidity")
+  
+  #Temperature, pressure, and relative humidity are required to calculate specific humidity.
+  humid_data = noaa_data[[humid_index]]
+  temperature_data = noaa_data[[which(cf_var_names == "air_temperature")]]
+  pressure_data = noaa_data[[which(cf_var_names == "air_pressure")]]
+   
+  #Depending on the volume and dimensions of data you download, sometimes R stores it as a vector and sometimes
+  #as a matrix; the different cases must be processed with different loops.
+  #(The specific corner case in which a vector would be generated is if only one hour is requested; for example, 
+  #only the data at time_idx 1, for example).
+  if (as.logical(nrow(humid_data))) {
+     for (i in 1:length(humid_data)) {
+       humid_data[i] = PEcAn.data.atmosphere::rh2qair(humid_data[i], temperature_data[i], pressure_data[i])
+     }
+  } else {
+     for (i in 1:nrow(humid_data)) {
+       for (j in 1:ncol(humid_data)) {
+          humid_data[i,j] = PEcAn.data.atmosphere::rh2qair(humid_data[i,j], temperature_data[i,j], pressure_data[i,j])
+        }
+      }
+  }
+  
+  #Update the noaa_data list with the correct data
+  noaa_data[[humid_index]] <- humid_data
+  
+  # Convert NOAA's total precipitation (kg m-2) to precipitation flux (kg m-2 s-1)
+  #NOAA precipitation data is an accumulation over 6 hours.
+  precip_index = which(cf_var_names == "precipitation_flux")
+  
+  #The statement udunits2::ud.convert(1, "kg m-2 6 hr-1", "kg m-2 s-1") is equivalent to udunits2::ud.convert(1, "kg m-2 hr-1", "kg m-2 s-1") * 6,
+  #which is a little unintuitive. What will do the conversion we want is what's below:
+  noaa_data[[precip_index]] = udunits2::ud.convert(noaa_data[[precip_index]], "kg m-2 hr-1", "kg m-2 6 s-1")  #There are 21600 seconds in 6 hours
+  
+  #############################################
+  # Done with data processing.  Now writing the data to the specified directory. Each ensemble member is written to its own file, for a total
+  # of 21 files.  
+  if (!dir.exists(outfolder)) {
+    dir.create(outfolder, recursive=TRUE)
+  }
```

Comment:
FWIW, `dir.create(outfolder, recursive = TRUE, showWarnings = FALSE)` will create the directory if it doesn't exist and succeed silently if it does (without `showWarnings = FALSE`, it will throw a warning but still succeed).